Doc/library/fcntl.rst

```diff
@@ -176,17 +175,17 @@ The module defines the following functions:
    method are accepted as well) of the file to lock or unlock, and *cmd*
    is one of the following values:
 
-   * :const:`LOCK_UN` -- unlock
-   * :const:`LOCK_SH` -- acquire a shared lock
-   * :const:`LOCK_EX` -- acquire an exclusive lock
+   * :const:`!LOCK_UN` -- unlock
+   * :const:`!LOCK_SH` -- acquire a shared lock
+   * :const:`!LOCK_EX` -- acquire an exclusive lock
 
-   When *cmd* is :const:`LOCK_SH` or :const:`LOCK_EX`, it can also be
-   bitwise ORed with :const:`LOCK_NB` to avoid blocking on lock acquisition.
-   If :const:`LOCK_NB` is used and the lock cannot be acquired, an
+   When *cmd* is :const:`!LOCK_SH` or :const:`!LOCK_EX`, it can also be
+   bitwise ORed with :const:`!LOCK_NB` to avoid blocking on lock acquisition.
+   If :const:`!LOCK_NB` is used and the lock cannot be acquired, an
    :exc:`OSError` will be raised and the exception will have an *errno*
-   attribute set to :const:`EACCES` or :const:`EAGAIN` (depending on the
+   attribute set to :data:`~errno.EACCES` or :data:`~errno.EAGAIN` (depending on the
```

Comment:
```suggestion
   attribute set to :const:`~errno.EACCES` or :const:`~errno.EAGAIN` (depending on the
```

`:const:` is equally syntactically valid and most semantically consistent and appropriate here, so no need to change the role type here.

Reply:
I changed it to `data` because `EAGAIN` and `EACESS` are defined in the `errno` module documentation using the `.. data::` directive.

Reply:
The [`.. py:data::` directive](https://www.sphinx-doc.org/en/master/usage/domains/python.html#directive-py-data) is used to define both constants (`:const:`) and (possibly variable) module data (`:data:`), presumably since their definitions are syntactically identical in Python and differ only in their usage semantics.

Also, in addition to the previously mentioned rationale (local consistency with other similar flags here, `const` being the most semantically appropriate choice as previously, and no reason to change the status quo), I also note that use of `:const:~ instead of `:data:` to refer to `errno.E` flags is standard convention across the docs, with 36 hits for `:const:`[~!]{0,2}errno\.E` and only 1 for `:data:`[~!]{0,2}errno\.E`.

Reply:
reverted my change